cmd/system-probe/subcommands/usm/symbols_ls_linux.go

```diff
@@ -0,0 +1,390 @@
+
+	// Parse version entries to build version index -> version string map
+	versionStrings := make(map[uint16]string)
+
+	// Parse verneed (version requirements - imported symbols)
+	if verneedSection != nil {
+		verneedData, err := verneedSection.Data()
+		if err == nil {
+			parseVerneed(verneedData, dynstrData, versionStrings)
+		}
+	}
+
+	// Parse verdef (version definitions - exported symbols)
+	if verdefSection != nil {
+		verdefData, err := verdefSection.Data()
+		if err == nil {
+			parseVerdef(verdefData, dynstrData, versionStrings)
+		}
+	}
+
+	// Map each symbol to its version
+	// Note: DynamicSymbols() skips the null symbol at index 0, but .gnu.version includes it
+	// So .gnu.version[0] is for null, .gnu.version[1] is for DynamicSymbols()[0], etc.
+	for i := 0; i < len(versionData)/sizeOfUint16; i++ {
+		// Each entry is 2 bytes (uint16)
+		versionIdx := uint16(versionData[i*sizeOfUint16]) | (uint16(versionData[i*sizeOfUint16+1]) << 8)
+
+		// Version index 0 and 1 are special (local and global)
+		if versionIdx > 1 {
+			// Clear the hidden bit (bit 15)
+			versionIdx &= 0x7fff
+			if verStr, ok := versionStrings[versionIdx]; ok {
+				// i=0 is null symbol, i=1 is DynamicSymbols()[0], etc.
+				if i > 0 {
+					versionMap[i-1] = verStr
+				}
+			}
+		}
+	}
+
+	return versionMap
+}
+
+// parseVerneed parses the .gnu.version_r section to extract version strings.
+func parseVerneed(verneedData, dynstrData []byte, versionStrings map[uint16]string) {
+	offset := 0
+	for offset < len(verneedData) {
+		if offset+16 > len(verneedData) {
+			break
+		}
+
+		// Read Verneed structure
+		cnt := binary.LittleEndian.Uint16(verneedData[offset+2:])
+		fileOffset := binary.LittleEndian.Uint32(verneedData[offset+4:])
+		aux := binary.LittleEndian.Uint32(verneedData[offset+8:])
+		next := binary.LittleEndian.Uint32(verneedData[offset+12:])
+
+		// Read the file name (library name) - not used in standard nm format
+		_ = readString(dynstrData, int(fileOffset))
+
+		// Parse auxiliary version entries
+		auxOffset := offset + int(aux)
+		for i := uint16(0); i < cnt; i++ {
+			if auxOffset+16 > len(verneedData) {
+				break
+			}
+
+			// Read Vernaux structure
+			other := binary.LittleEndian.Uint16(verneedData[auxOffset+6:])
+			nameOffset := binary.LittleEndian.Uint32(verneedData[auxOffset+8:])
+			nextAux := binary.LittleEndian.Uint32(verneedData[auxOffset+12:])
+
+			// Read version name
+			versionName := readString(dynstrData, int(nameOffset))
+
+			// Store the version string with @ prefix (standard nm format)
+			versionStrings[other] = "@" + versionName
+
+			if nextAux == 0 {
+				break
+			}
+			auxOffset += int(nextAux)
+		}
+
+		if next == 0 {
+			break
+		}
+		offset += int(next)
+	}
+}
+
+// parseVerdef parses the .gnu.version_d section to extract version definitions.
+func parseVerdef(verdefData, dynstrData []byte, versionStrings map[uint16]string) {
+	offset := 0
+	for offset < len(verdefData) {
+		if offset+20 > len(verdefData) {
```

Comment:
NIT: why 20? 
maybe make one more iteration on consts in the changes

Reply:
as we read the first 20 characters, we need to verify we have at least 20 characters to read